src/panels/lovelace/components/hui-buttons-base.ts

```diff
@@ -46,22 +50,32 @@ export class HuiButtonsBase extends LitElement {
         }
 
         return html`
-          <div>
-            <state-badge
-              title=${computeTooltip(this._hass!, entityConf)}
-              @action=${this._handleAction}
-              .actionHandler=${actionHandler({
-                hasHold: hasAction(entityConf.hold_action),
-                hasDoubleClick: hasAction(entityConf.double_tap_action),
-              })}
-              .config=${entityConf}
-              .hass=${this._hass}
-              .stateObj=${stateObj}
-              .overrideIcon=${entityConf.icon}
-              .overrideImage=${entityConf.image}
-              stateColor
-              tabindex="0"
-            ></state-badge>
+          <div
+            @action=${this._handleAction}
+            .actionHandler=${actionHandler({
+              hasHold: hasAction(entityConf.hold_action),
+              hasDoubleClick: hasAction(entityConf.double_tap_action),
+            })}
+            .config=${entityConf}
+            tabindex="0"
+          >
+            ${entityConf.show_icon !== false
+              ? html`
+                  <state-badge
+                    title=${computeTooltip(this._hass!, entityConf)}
+                    _hass=${this._hass}
+                    .stateObj=${stateObj}
+                    .overrideIcon=${entityConf.icon}
+                    .overrideImage=${entityConf.image}
+                    stateColor
+                  ></state-badge>
+                `
+              : ""}
+            <span>
+              ${entityConf.show_name || entityConf.name
```

Comment:
Shouldn't this be just `entityConf.show_name`? I see what you were going for but I think it should default off and only show if show name is specified as true

Reply:
That seems silly, you have to manually add `name` and then also set `show_name: true`?

Reply:
Then show name should be defaulted to true

Reply:
Example config
```yaml
entity: entity.light
name: This is a light
show_name: false
```
This would still show the name they chose and that is incorrect in my opinion

Reply:
perhaps a compromise then
```
entityConf.show_name || (entityConf.show_name !== false && entityConf.name)
```

Reply:
That will work 👍

Reply:
Although with an editor, it will be false. I'm relying on it being undefined at the moment 🤷‍♂